src/wms/src/test/java/org/geoserver/wms/WMSVectorDimensionValidationTest.java

```diff
@@ -0,0 +1,564 @@
+import java.util.List;
+import java.util.Map;
+import javax.xml.namespace.QName;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.DimensionInfo;
+import org.geoserver.catalog.DimensionPresentation;
+import org.geoserver.catalog.FeatureTypeInfo;
+import org.geoserver.catalog.impl.DimensionInfoImpl;
+import org.geoserver.data.test.SystemTestData;
+import org.geoserver.platform.ServiceException;
+import org.geotools.api.data.FeatureSource;
+import org.geotools.api.data.Query;
+import org.geotools.referencing.crs.DefaultGeographicCRS;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.locationtech.jts.geom.Envelope;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/**
+ * This suite performs tests on the WMS request validation feature added for <a
+ * href="https://osgeo-org.atlassian.net/browse/GEOS-9757">GEOS-9757 - Return a service exception
+ * when client provided WMS dimensions are not a match</a>.
+ *
+ * @author skalesse
+ * @since 2024-07-11
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class WMSVectorDimensionValidationTest extends WMSTestSupport {
+
+    private static final QName TIME_WITH_START_END =
+            new QName(SF_URI, "TimeWithStartEnd", SF_PREFIX);
+    private static final QName ELEVATION_WITH_START_END =
+            new QName(SF_URI, "ElevationWithStartEnd", SF_PREFIX);
+    private static final QName CUSTOM_DIM_WITH_START_END =
+            new QName(SF_URI, "CustomDimensionWithStartEnd", SF_PREFIX);
+
+    private WMS wms;
+
+    @Before
+    public void setWMS() {
+        wms = getWMS();
+    }
+
+    @Override
+    protected void onSetUp(SystemTestData testData) throws Exception {
+        super.onSetUp(testData);
+        String testDataFileName = "TimeElevationWithStartEnd.properties";
+        testData.addVectorLayer(
+                TIME_WITH_START_END, emptyMap(), testDataFileName, getClass(), getCatalog());
+
+        testData.addVectorLayer(
+                ELEVATION_WITH_START_END, emptyMap(), testDataFileName, getClass(), getCatalog());
+
+        testData.addVectorLayer(
+                CUSTOM_DIM_WITH_START_END, emptyMap(), testDataFileName, getClass(), getCatalog());
+    }
+
+    /**
+     * For a 'time' dimension, if a matching value is provided in the request the validation should
+     * succeed
+     *
+     * @throws IOException an unexpected IO exception is thrown
+     */
+    @Test
+    public void matchingTimeDimensionShouldValidate() throws IOException {
+        // the dimension and its value
+        String dimension = "time";
+        Date dimensionValue = Date.from(Instant.parse("2012-02-11T10:15:30.00Z"));
+        // create a map request
+        GetMapRequest request = mapRequest(singletonList(dimensionValue), emptyList(), emptyMap());
+        // disable validation
+        setValidationEnabled(true);
```

Comment:
```suggestion
        // enable validation
        setValidationEnabled(true);
```

The same lines have been copied over, so is the inconsistency between comment and code is found in other test methods as well.

Reply:
Ups. Thanks for finding those! I fixed two places where there where inconsistent comments.